pittapi/gym.py

```diff
@@ -0,0 +1,86 @@
+"""
+The Pitt API, to access workable data of the University of Pittsburgh
+Copyright (C) 2015 Ritwik Gupta
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+
+from __future__ import annotations
+
+from bs4 import BeautifulSoup
+import requests
+from typing import NamedTuple
+
+GYM_URL = "https://connect2concepts.com/connect2/?type=bar&key=17c2cbcb-ec92-4178-a5f5-c4860330aea0"
+
+GYM_NAMES = [
+    "Baierl Rec Center",
+    "Bellefield Hall: Fitness Center & Weight Room",
+    "Bellefield Hall: Court & Dance Studio",
+    "Trees Hall: Fitness Center",
+    "Trees Hall: Courts",
+    "Trees Hall: Racquetball Courts & Multipurpose Room",
+    "William Pitt Union",
+    "Pitt Sports Dome",
+]
+
+
+class Gym(NamedTuple):
+    name: str
+    date: str
+    count: int
+    percentage: int
+
+    @classmethod
+    def from_text(cls, text: str) -> "Gym":
```

Comment:
```suggestion
from typing import NamedTuple, Self

GYM_URL = "https://connect2concepts.com/connect2/?type=bar&key=17c2cbcb-ec92-4178-a5f5-c4860330aea0"

GYM_NAMES = [
    "Baierl Rec Center",
    "Bellefield Hall: Fitness Center & Weight Room",
    "Bellefield Hall: Court & Dance Studio",
    "Trees Hall: Fitness Center",
    "Trees Hall: Courts",
    "Trees Hall: Racquetball Courts & Multipurpose Room",
    "William Pitt Union",
    "Pitt Sports Dome",
]


class Gym(NamedTuple):
    name: str
    date: str
    count: int
    percentage: int

    @classmethod
    def from_text(cls, text: str) -> Self:
```
I see that you annotated the return type for `from_text` as `"Gym"`. FYI, this isn't wrong and does work correctly, but I think a slightly better solution would be to use the [`Self` type from the `typing` module](https://docs.python.org/3/library/typing.html#typing.Self), which allows us to specify the type without directly referencing the class name.

Reply:
NVM, seems like that feature isn't supported in `typing` until 3.11 😓